README.md

```diff
@@ -1,102 +1,120 @@
+* It initializes the AIDE database
+* The AIDE databases from the remote nodes are stored in a central directory on the controller node
+* It runs AIDE integrity checks on the remote nodes
+* It updates the AIDE databases and stores them on the controller node
 
-For instance, if the role depends on some collections and has a
-`meta/collection-requirements.yml` file for installing those dependencies, and
-in order to manage `rpm-ostree` systems, it should be mentioned here that the
- user should run
+## How does the role do that?
 
-```bash
-ansible-galaxy collection install -vv -r meta/collection-requirements.yml
-```
+* The role is controlled by using role variables
+* If you run the playbook without specifying any role variable the role will change nothing on your remote nodes
+* To execute some supported use cases you need to explicitly specify one or more of the following variables
+
+### Available tags to control and use the role
 
-on the *control node* before using the role.
+## What does this role not do for you?
+
+* It does not explain how to create a good AIDE configuration that suits your requirements; that task remains for you to accomplish
+
+## Requirements
+
+This role has no special requirements as it uses `ansible.builtin` modules
+only.
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.  Each variable should have its own section e.g.
+### aide_db_fetch_dir
 
-### aide_foo
+This variable takes a string to specify the directory on the Ansible Control
+Node (ACN) where the role will store the AIDE database fetched from the remote
+nodes. The default value is `files` which is expected to be a directory in the
+same directory as the playbook.
 
-This variable is required.  It is a string that lists the foo of the role.
-There is no default value.
+In case you like to store the fetched AIDE database files somewhere else you
+need to specify a different path here.
 
-### aide_bar
+### aide_install
 
-This variable is optional.  It is a boolean that tells the role to disable bar.
-The default value is `true`.
+With this variable the role ensures that the `aide` package is installed on the remote nodes
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+Default: `false`
 
-Example of setting the variables:
+Type: `bool`
 
-```yaml
-aide_foo: "oof"
-aide_bar: false
-```
+### aide_generate_config
+
+Generates the file `/etc/aide.conf` using `templates/aide.conf.j2`; the template needs to be adjusted to fit your requirements; if you do not use this varable the default configuration file shipped with the `aide` package will be used.
```

Comment:
> the template needs to be adjusted to fit your requirements;

What does this mean?  Note that, in general, users will not be able to modify the template file.  If we really want users to be able to provide custom configuration, then we should do either/both of these things:
1) Users can provide a list of configuration parameters that are written in the template
2) Users can provide their own entire `aide.conf` file

Reply:
> > the template needs to be adjusted to fit your requirements;
> 
> What does this mean? Note that, in general, users will not be able to modify the template file. Was this in the original role? If so, then I guess that use case was different i.e. as a consultant, I need to install this role in a writable directory on the customers machine and modify and run the role as I require for my particular use case. This is definitely not the linux system roles use case, where the only thing the user may do is modify the inventory and the playbook, and provide files/templates - nothing else.
> 
> If we really want users to be able to provide custom configuration, then we should do either/both of these things:
> 
>     1. Users can provide a list of configuration parameters that are written in the template
> 
>     2. Users can provide their own `aide.conf` file or template

Yeah, as you pointed out, this was designed originally for different use case. Let me evaluate options.

Reply:
In case user provides aide.conf should we prepend ansible_managed comment?